hamster_gtk/misc/widgets/raw_fact_entry.py

```diff
@@ -0,0 +1,127 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# 'hamster-gtk' is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with 'hamster-gtk'.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Widget meant to handle 'raw-fact' strings and provides autocompletion."""
+from __future__ import absolute_import, unicode_literals
+import datetime
+
+from gi.repository import GObject, Gtk
+from orderedset import OrderedSet
+from six import text_type
+from hamster_lib import Category, Fact
+from hamster_gtk.helpers import _u
+from hamster_gtk import helpers
+
+
+class RawFactEntry(Gtk.Entry):
+    """A custom entry widgets that provides ``faw fact`` specific autocompletion behaviour."""
+
+    def __init__(self, controller, *args, **kwargs):
+        """Instantiate the class."""
+        super(RawFactEntry, self).__init__(*args, **kwargs)
+        self._controller = controller
+        self._controller.signal_handler.connect('facts-changed', self._on_facts_changed)
+        self.set_completion(RawFactCompletion(self._controller))
+
+    # Callbacks
+    def _on_facts_changed(self, evtl):
+        """Callback triggered when facts have changed."""
+        self.set_completion(RawFactCompletion(self._controller))
+
+
+class RawFactCompletion(Gtk.EntryCompletion):
+    """
+    Return a completion instance to match 'activity@category' strings.
+
+    Returns:
+        Gtk.EntryCompletion: Completion instance.
+    """
+
+    def __init__(self, controller, *args, **kwargs):
+        """instantiate class."""
+        super(RawFactCompletion, self).__init__(*args, **kwargs)
+        self._controller = controller
+        self.set_model(self._get_store())
+        self.set_text_column(0)
+        self.set_match_func(self._match_anywhere, None)
+        self.connect('match-selected', self._on_match_selected)
+
+    def _get_store(self):
+        store = Gtk.ListStore(GObject.TYPE_STRING, GObject.TYPE_STRING, GObject.TYPE_STRING)
+        for activity in self._get_activities():
+            category = ''
+            if activity.category:
+                category = text_type(activity.category.name)
+            store.append([
+                helpers.serialise_activity(activity), text_type(activity.name), category
+            ])
+        return store
+
+    def _get_activities(self):
+        today = datetime.date.today()
+        recent_activities = [fact.activity for fact in self._controller.facts.get_all(
+            start=today, end=today)]
+        return OrderedSet(recent_activities)
+
+    def _match_anywhere(self, entrystr, iter, data, *args):
```

Comment:
Actually you are missing the `completion` attribute (see the reference). It should be like this:
```python
def _match_anywhere(self, completion, entrystr, iter, data):
```